DOC: Better explanations about GPGPU driver installations (#2464)
uxlfoundation/scikit-learn-intelex: doc/sources/oneapi-gpu.rst

oneAPI and GPU support in |sklearnex|
##############################################################

> |sklearnex| can execute computations on different devices (CPUs and GPUs, including integrated GPUs from laptops and desktops) through the SYCL framework in oneAPI.

Reviewer: Same here maybe just say ((CPUs and GPUs) instead of (CPUs and GPUs, including integrated GPUs from laptops and desktops)

Reviewer 2: Agreed

The device used for computations can be easily controlled through the target offloading functionality (e.g. through ``sklearnex.config_context(target_offload="gpu")``, which moves data to GPU if it's not already there - see rest of this page for more details), but for finer-grained controlled (e.g. operating on arrays that are already in a given device's memory), it can also interact with objects from package |dpctl|, which offers a Python interface over SYCL concepts such as devices, queues, and USM (unified shared memory) arrays.

> While not strictly required, package |dpctl| is recommended for a better experience on GPUs - for example, it can provide GPU-allocated arrays so that ``target_offload`` wouldn't need to move the data from CPU to GPU.

Reviewer 3: As an option you can use "compute follows data" name to describe it as well.

Author: Reworded.

.. important:: Be aware that GPU usage requires non-Python dependencies on your system, such as the `Intel(R) GPGPU Drivers <https://www.intel.com/content/www/us/en/developer/articles/system-requirements/intel-oneapi-dpcpp-system-requirements.html>`_ (see below).

Prerequisites
-------------

For execution on GPUs, DPC++ runtime and GPGPU drivers are required.

> DPC++

Reviewer 3: Might worth to replace it to SYCL

Author: But it does require the software called "DPC++ runtime":
https://anaconda.org/conda-forge/dpcpp-cpp-rt

> DPCP++

Reviewer 4: ```suggestion
DPC++
```

     conda install -c conda-forge dpcpp_cpp_rt

GPGPU Drivers
~~~~~~~~~~~~~

> On windows, GPGPU drivers are bundled as part of the regular drivers for iGPUs and dGPUs, which can be downloaded from `this link <https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html>`__.

Reviewer 4: ```suggestion
On Windows, GPGPU drivers are bundled as part of the regular drivers for iGPUs and dGPUs, which can be downloaded from `this link <https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html>`__.
```

     conda install -c conda-forge dpcpp_cpp_rt

GPGPU Drivers
~~~~~~~~~~~~~

> On Windows, GPGPU drivers are bundled as part of the regular drivers for iGPUs and dGPUs, which can be downloaded from `this link <https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html>`__.

Reviewer: I notice here you have GPGPU was that a typo?

> On Windows, GPGPU drivers are bundled as part of the regular drivers for iGPUs and dGPUs, which can be downloaded from `this link <https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html>`__.

Author: I think that wouldn't be very obvious to people who are not familiar with the product suite already. Before joining Intel, I wouldn't have known that laptop iGPUs were called "Iris Xe", for example.

For datacenters, see further instructions `here <https://www.intel.com/content/www/us/en/developer/articles/system-requirements/oneapi-dpcpp/2025.html#inpage-nav-2-1-1>`__.

> On Linux, some distributions - namely Ubuntu Desktop 25.04 and higher, and Fedora Workstation 42 and higher - come with the GPGPU drivers for iGPUs and dGPUs preinstalled, while others require installing them separately.

Reviewer: samething maybe just say gpu instead of GPGPU

On Debian systems, it requires installing package ``intel-opencl-icd``, which is available from Debian's ``main`` repository: ::

    sudo apt-get install intel-opencl-icd

> For Arch Linux, and for other distributions in general, see the `GPGPU article in the Arch wiki <https://wiki.archlinux.org/title/GPGPU>`__.

Author: Problem with that page are that: (a) it doesn't explain how to install the drivers in most of the supported platforms; (b) the drivers and processes it lists are for datacenter GPUs.

     conda install -c conda-forge dpcpp_cpp_rt

Intel Compute Runtime
~~~~~~~~~~~~~~~~~~~~~

On Windows, GPU drivers for iGPUs and dGPUs include the required Intel Compute Runtime. Drivers for windows can be downloaded from `this link <https://www.intel.com/content/www/us/en/download/785597/intel-arc-iris-xe-graphics-windows.html>`__.

For datacenters, see further instructions `here <https://www.intel.com/content/www/us/en/developer/articles/system-requirements/oneapi-dpcpp/2025.html#inpage-nav-2-1-1>`__.

On Linux, some distributions - namely Ubuntu Desktop 25.04 and higher, and Fedora Workstation 42 and higher - come with the compute runtime for iGPUs and dGPUs preinstalled, while others require installing them separately.

> On Debian systems, it requires installing package ``intel-opencl-icd`` (along with its dependencies such as ``intel-compute-runtime`` and ``intel-graphics-compiler``), which is available from Debian's ``main`` repository: ::

Reviewer 2: ```suggestion
Debian systems require installing package ``intel-opencl-icd`` (along with its dependencies such as ``intel-compute-runtime`` and ``intel-graphics-compiler``), which is available from Debian's ``main`` repository: ::
```

    sudo apt-get install intel-opencl-icd

> .. tip:: For Debian Trixie (13), the Intel Compute Runtime is not available from the Stable repository, but can be installed by enabling the Sid (Unstable) repository.

Reviewer 2: Are stable and unstable actual names/proper nouns here or should they be lowercase?

Author: They aren't, but that kind of usage is common throughout docs:
https://wiki.debian.org/SourcesList
